src/app/services/courses.service.ts

```diff
@@ -1,44 +1,61 @@
 import { Injectable } from '@angular/core';
+import { Course } from '@shared/models/course.model';
+import { HttpClient } from '@angular/common/http';
+import { Author } from '@shared/models/author.model';
+import { Observable } from 'rxjs';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CoursesService {
-  getAll() {
-    // Add your code here
+  private API_URL = 'http://localhost:4000';
+
+  constructor(private http: HttpClient) {}
+
+  getAll(): Observable<{ result: Course[] }> {
+    return this.http.get<{ result: Course[] }>(`${this.API_URL}/courses/all`);
   }
 
-  createCourse(course: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  createCourse(course: Course) {
+    return this.http.post<Course>(`${this.API_URL}/courses/add`, course);
   }
 
-  editCourse(id: string, course: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  editCourse(id: string, course: Course): Observable<{ result: Course }> {
+    return this.http.put<{ result: Course }>(
+      `${this.API_URL}/courses/${id}`,
+      course
+    );
   }
 
-  getCourse(id: string) {
-    // Add your code here
+  getCourse(id: string): Observable<{ result: Course }> {
+    return this.http.get<{ result: Course }>(`${this.API_URL}/courses/${id}`);
   }
 
-  deleteCourse(id: string) {
+  deleteCourse(id: string): Observable<{ result: Course }> {
     // Add your code here
+    return this.http.delete<{ result: Course }>(
```

Comment:
I dont think the delete method will have a return type, definitely not a Course. Maybe void...

Reply:
i rewrote it